### DISCUSSION TITLE
Rebuilding node_modules?

### DISCUSSION THREAD

**Discussion by RedTn** [yarnpkg/berry#4877] [General]
What's the best way to rebuild my node_modules folder?
Context: sometimes I like to mess around manually in there, maybe typing .js files manually inside node_modules folder.

Would it be to run:
- yarn clean
- yarn install
??

I find this method is slow, back in yarn v1 I would use https://classic.yarnpkg.com/lang/en/docs/cli/install/#toc-yarn-install-check-files

**Comment by fictitious** [comment]
I don't know anything better than removing `node_modules` and removing `.yarn/install-state.tgz`, then doing `yarn install`. Yes it's slow.

I made [yarn plugin](https://github.com/fictitious/yarn-plugins#yarn-plugin-clean-state) just for that - after adding the plugin, you can run `yarn clean state` then `yarn install`.

**Comment by jj811208** [comment]
Maybe you could try the official plugin
https://github.com/yarnpkg/berry/tree/master/packages/plugin-patch and just delete it when you want to go back to the original, which I think is much faster than rebuilding node_modules.

**Comment by belgattitude** [comment]
I tend to just add a script in package.json for the extremely rare occasions. I don't remove the install-state as it does not seem to create issues.

```json5
{
  "scripts": {
    // The glob **/ is for monorepos
     "nuke:node_modules": "rimraf '**/node_modules'",
   },
  "devDependencies": {
     "rimraf": "3.0.2",
   }
}
```

Just run `yarn nuke:node_modules` followed by a `yarn install`.
